src/main/java/org/candlepin/pki/impl/BouncyCastlePKIUtility.java

```diff
@@ -79,45 +78,50 @@
 import java.util.Date;
 import java.util.Set;
 
+import javax.inject.Provider;
+
 /**
  * The default {@link ProviderBasedPKIUtility} for Candlepin.
  * This class implements methods to create X509 Certificates, X509 CRLs, encode
  * objects in PEM format (for saving to the db or sending to the client), and
  * decode raw ASN.1 DER values (as read from a Certificate/CRL).
- *
+ * <p>
  * All code that imports bouncycastle should live in this module.
- *
+ * <p>
  * (March 24, 2011) Notes on implementing a ProviderBasedPKIUtility with NSS/JSS:
- *
+ * <p>
  * JSS provides classes and functions to generate X509Certificates (see CertificateInfo,
  * for example).
- *
+ * <p>
  * PEM encoding requires us to determine the object type (which we know), add the correct
  * header and footer to the output, base64 encode the DER for the object, and line wrap
  * the base64 encoding.
- *
+ * <p>
  * decodeDERValue should be simple, as JSS provides code to parse ASN.1, but I wasn't
  * able to get it to work.
- *
+ * <p>
  * The big one is CRL generation. JSS has no code to generate CRLs in any format. We'll
  * have to use the raw ASN.1 libraries to build up our own properly formatted CRL DER
  * representation, then PEM encode it.
  */
 public class BouncyCastlePKIUtility extends ProviderBasedPKIUtility {
-    private static Logger log = LoggerFactory.getLogger(BouncyCastlePKIUtility.class);
-    public static final byte[] LINE_SEPARATOR = String.format("%n").getBytes();
-    public static final String SIGNING_ALG_ID = "SHA256withRSA";
+    private static final Logger log = LoggerFactory.getLogger(BouncyCastlePKIUtility.class);
+    private static final byte[] LINE_SEPARATOR = String.format("%n").getBytes();
+    private static final String SIGNING_ALG_ID = "SHA256withRSA";
+    private static final String KEY_ALGORITHM = "RSA";
+    private static final int KEY_SIZE = 4096;
+    private static final String PRIVATE_KEY_PEM_NAME = "PRIVATE KEY";
 
+    private final Provider<BouncyCastleProvider> securityProvider;
     private final KeyPairDataCurator keypairDataCurator;
-    private static final String KEY_ALGORITHM = "RSA";
-    public static final int KEY_SIZE = 4096;
-    public static final String PRIVATE_KEY_PEM_NAME = "PRIVATE KEY";
 
     @Inject
-    public BouncyCastlePKIUtility(CertificateReader reader, SubjectKeyIdentifierWriter subjectKeyWriter,
-        Configuration config, KeyPairDataCurator keypairDataCurator) {
+    public BouncyCastlePKIUtility(Provider<BouncyCastleProvider> securityProvider, CertificateReader reader,
+        SubjectKeyIdentifierWriter subjectKeyWriter, Configuration config,
+        KeyPairDataCurator keypairDataCurator) {
         super(reader, subjectKeyWriter, config);
         this.keypairDataCurator = keypairDataCurator;
+        this.securityProvider = securityProvider;
```

Comment:
Can we use the opportunity to add the null checks on our dependencies here? I wonder if we have some old tests which currently throw nulls at this.

Reply:
Done.